lib/include/net/cne_net.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright 2016 6WIND S.A.
+ */
+
+#ifndef _CNE_NET_H_
+#define _CNE_NET_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file cne_net.h
+ *
+ * Network checksum functions to prepare pseudo header checksum for TSO and non-TSO
+ * TCP/UDP data.
+ */
+
+#include <net/cne_ip.h>
+#include <net/cne_udp.h>
+#include <net/cne_tcp.h>
+
+/**
+ * Prepare pseudo header checksum
+ *
+ * This function prepares pseudo header checksum for TSO and non-TSO tcp/udp in
+ * provided mbufs packet data and based on the requested offload flags.
+ *
+ * - for non-TSO tcp/udp packets full pseudo-header checksum is counted and set
+ *   in packet data,
+ * - for TSO the IP payload length is not included in pseudo header.
+ *
+ * This function expects that used headers are in the first data segment of
+ * mbuf, are not fragmented and can be safely modified.
+ *
+ * @param m
+ *   The packet mbuf to be fixed.
+ * @param ol_flags
+ *   TX offloads flags to use with this packet.
+ * @return
+ *   0 if checksum is initialized properly
+ */
+static inline int
+cne_net_intel_cksum_flags_prepare(pktmbuf_t *m, uint64_t ol_flags)
```

Comment:
Why does this API have "intel" in the name?

Reply:
This code came from DPDK some time back, but will remove intel from the names.